modules/library/main/src/main/java/org/geotools/geometry/jts/GeometryClipper.java

```diff
@@ -76,6 +79,49 @@ public GeometryClipper(Envelope bounds) {
     }
     
     /**
+     * This will try to handle failures when clipping - i.e. because of invalid input geometries (often 
+     * caused by simplification).
+     * 
+     * This attempts to do a normal clip().  If it fails, it will try to do more to ensure the clip 
+     * works properly.  
+     * 
+     * The first attempt to correct the geometry is to put the points on a precision grid and redo
+     * the clip. 
+     * If this fails, the original geometry is returned.
+     * 
+     * see {@link #clip(Geometry, boolean) clip}
+     * 
+     * @param g
+     * @param ensureValid
+     * @param scale 0=double precision Precision Model
+     * @return
+     */
+    public Geometry clipFailResistant(Geometry g, boolean ensureValid, double scale) {
```

Comment:
Hum... hum... not robust either... it's more like "clip if you can, don't if you don't". It made sense in the context of streaming renderer, where not clipping is not always fatal (it is only if they line is dashed and very long compared to the viewing area, an instant OOM generator due to a JDK bug), not sure it does make sense as a public method though. @jodygarnett , what do you think?